### PR TITLE
Add pinging

### PR DESCRIPTION
**Pull request by jmullan**

### CODE REVIEW

File: sutrofm/api_views.py [hunk 1]
```diff
@@ -8,6 +9,8 @@
 
 from sutrofm.redis_models import Message, Party, User
 
+logger = logging.getLogger(__name__)
```
Comment (superemily) [inline sutrofm/api_views.py]: you're addicted to logger, i think you have a problem

File: sutrofm/api_views.py [hunk 2]
```diff
@@ -164,6 +169,24 @@ def ping(request):
 
 
 @csrf_exempt
+def ping_party(request, party_id):
+  logger.info('r %r', request.__dict__)
```
Comment (mkapolka) [inline sutrofm/api_views.py]: This doesn't seem useful and will get hit a whole lot

File: sutrofm/api_views.py [hunk 3]
```diff
@@ -164,6 +169,24 @@ def ping(request):
...
+  redis = StrictRedis(connection_pool=redis_connection_pool)
+  user = User.from_request(redis, request)
+  focused = False
+  if user:
+    user.last_check_in = datetime.datetime.utcnow()
+    user.save(redis)
+    party = Party.get(redis, party_id)
+    if party:
+      party.add_user(redis, user, focused)
+      party.save(redis)
```
Comment (mkapolka) [inline sutrofm/api_views.py]: Would be nice to only do this if we actually add the user

File: sutrofm/redis_models.py [hunk 4]
```diff
@@ -431,6 +429,48 @@ def from_request(connection, request):
...
+  def seen_rooms(self, connection):
+    return connection.smembers('user:%s:parties' % self.id)
+
+  def active_rooms(self, connection):
+    return [
+      party_id for party_id in self.seen_rooms(connection)
+      if self.is_active(connection, party_id)
+    ]
+
+  def focused_room(self, connection):
```
Comment (mkapolka) [inline sutrofm/redis_models.py]: NO

**Comment by mkapolka** [comment]
Patch does not meet pun requirements. Please add more references to i.e. coconut technology, Gilligan's island, etc. Please refer to our community silliness guidelines here: http://github.com/superemily/sutrofm/wiki/pages/420

**Pull request merged**